app/models/access_token.rb

```diff
@@ -0,0 +1,17 @@
+class AccessToken < ApplicationRecord
+  validates :token, presence: true, uniqueness: true
+  
+  belongs_to :user
+
+  after_initialize :generate_token
+
+  private
+
+  def generate_token
+    loop do
+      break if token.present? && !AccessToken.exists?(token: token)
+      self.token = SecureRandom.hex(10)
+    end
```

Comment:
Unsure of the loop implemented here.  Done as suggested by the instructor for the lesson.